src/operators/convert.ts

```diff
@@ -83,6 +83,8 @@ export class ConvertOperator implements Operator {
   handleData(data: any[]): any[] | null {
     // NOTE this operator transforms data - be absolutely sure there are no side effects to the data layer!
 
+    const index = this.index >= 0 ? this.index : data.length + this.index;
```

Comment:
This is really the main change.  The lines below have find/replace on `this.index` with `index`.